android/java/org/chromium/chrome/browser/crypto_wallet/util/ImageLoader.java

```diff
@@ -51,6 +45,6 @@ public static boolean isSvg(String url) {
     }
 
     private static boolean isValidImgUrl(String url) {
-        return !TextUtils.isEmpty(url);
+        return URLUtil.isDataUrl(url) || URLUtil.isNetworkUrl(url);
```

Comment:
Non blocker: We also support IPFS images. Do we have access to an IPFS node directly on android enabled by the IPFS flag? If not all good. If we wanted to still support them we could accept `ipfs://` URLs by translating them into a default gateway URL and resolving the content from a gateway. Up to you if you want to do this though.

Reply:
During parsing, we replace `ipfs` urls like desktop [trimedUrl.replace("ipfs://", "https://ipfs.io/ipfs/")](https://github.com/brave/brave-core/blob/5995c24e3d82297e5ce3c50968b2501c6ada4bcd/android/java/org/chromium/chrome/browser/crypto_wallet/util/AssetUtils.java#L178) so this should be okay?

Reply:
At the moment we are _httpifying_ IPFS URLS. Meaning when an URL starts with `ipfs://` we replace it with `https://ipfs.io/ipfs/`. The util method in charge of this has been defined here:
https://github.com/brave/brave-core/blob/a3503ed28bd810086cf930f355be0d34ba88770f/android/java/org/chromium/chrome/browser/crypto_wallet/util/AssetUtils.java#L188-L194

I've adopted this logic by looking at the iOS implementation.

Reply:
Lgtm